ravenframework/CodeInterfaceClasses/MELCOR/melcorCombinedInterface.py

```diff
@@ -13,104 +13,178 @@
+
+    return currentInputFiles
+
+  def writeDict(self,workDir):
+    """
+      Output the parsed results into a CSV file
+      @ In, workDir, str, current working directory
+      @ Out, dictionary, dict, dictioanry containing the data generated by MELCOR
+    """
+    fileDir = os.path.join(workDir,self.melcorPlotFile)
+    print(self.varList)
+    time,data,varUdm = melcorTools.MCRBin(fileDir,self.varList)
+    dfTime = pd.DataFrame(time, columns= ["Time"])
+    dfData = pd.DataFrame(data, columns = self.varList)
+    df = pd.concat([dfTime, dfData], axis=1, join='inner')
+    df.drop_duplicates(subset="Time",keep='first',inplace=True)
+    dictionary = df.to_dict(orient='list')
+    return dictionary
+
+  def finalizeCodeOutput(self,command,output,workingDir):
     """
       This method is called by the RAVEN code at the end of each run (if the method is present, since it is optional).
-      In this method the MELCOR outputfile is parsed and a CSV is created
+      It can be used for those codes, that do not create CSV files to convert the whatever output format into a csv
       @ In, command, string, the command used to run the just ended job
       @ In, output, string, the Output name root
       @ In, workingDir, string, current working dir
-      @ Out, output, string, optional, present in case the root of the output file gets changed in this method.
+      @ Out, response, dict, dictionary containing the data
     """
-    output = self.melcorInterface.finalizeCodeOutput(command,output, workingDir)
-    return output
+    response = self.writeDict(workingDir)
+    return response
+
 
   def checkForOutputFailure(self,output,workingDir):
     """
       This method is called by the RAVEN code at the end of each run  if the return code is == 0.
       This method needs to be implemented by the codes that, if the run fails, return a return code that is 0
       This can happen in those codes that record the failure of the job (e.g. not converged, etc.) as normal termination (returncode == 0)
       This method can be used, for example, to parse the outputfile looking for a special keyword that testifies that a particular job got failed
-      (e.g. in MELCOR would be the expression "Normal termination")
+      (e.g. in RELAP5 would be the keyword "********")
       @ In, output, string, the Output name root
       @ In, workingDir, string, current working dir
       @ Out, failure, bool, True if the job is failed, False otherwise
     """
-    failure = self.melcorInterface.checkForOutputFailure(output, workingDir)
+    failure = True
+    goodWord  = "Normal termination"   # This is for MELCOR 2.2 (todo: list for other MELCOR versions)
+    try:
+      outputToRead = open(os.path.join(workingDir,self.melcorOutFile),"r")
+    except FileNotFoundError:
+      return failure
+    readLines = outputToRead.readlines()
+    lastRow = readLines[-1]
+    if goodWord in lastRow:
+      failure = False
+    outputToRead.close()
```

Comment:
if you return early (in the try except) this file is never closed. 

I would use the construct 
```
   with open(os.path.join(workingDir,self.melcorOutFile),"r") as outputToRead:

    ------ operations here ----
```